docs/features/bgp-integration/route-advertisements.md

```diff
@@ -0,0 +1,620 @@
+# Route Advertisements
+
+## Introduction
+
+The Route Advertisements feature introduces BGP as a supported routing protocol
+with OVN-Kubernetes enabling the integration into different BGP user
+environments. The extent of the Route Advertisements feature and corresponding
+API allows importing routes from BGP peers into OVN pod networks as well as
+exporting pod network and egress IP routes to BGP peers. Both default pod
+network as well as primary Layer 3 and Layer 2 cluster-user-defined networks
+(CUDNs) are supported.
+
+## Prerequisites
+
+- [FRR-k8s](https://github.com/metallb/frr-k8s)
+
+## Motivation
+
+There are multiple driving factors which necessitate integrating BGP into
+OVN-Kubernetes (for purposes of this documentation, the external, physical
+network of the cluster which a user administers will be called the “provider
+network”):
+
+- Importing Routes from the Provider Network: Today there is no API for a user
+to be able to configure routes into OVN. In order for a user to change how
+egress traffic is routed, the user leverages local gateway mode. This mode
+forces traffic to hop through the Linux networking stack, and there a user can
+configure routes inside the host to control egress routing. This manual
+configuration would need to be performed and maintained across nodes and VRFs
+within each node.
+
+- Exporting Routes into the Provider Network: There exists a need for provider
+networks to learn routes directly to services and pods today in Kubernetes. One
+such use case is integration with 3rd party load balancers, where they terminate
+a load balancer and then send packets directly to cluster nodes with the
+destination IP address being the pod IP itself. Today these load balancers rely
+on custom operators to detect which node a pod is scheduled to and then add
+routes into its load balancer to send the packet to the right node. By
+integrating BGP and advertising the pod subnets/addresses directly on the
+provider network, load balancers and other entities on the network would be able
+to reach the pod IPs directly.
+
+Additionally, integrating BGP support paves the way for other BGP based features
+that might be implemented in the future, like:
+
+- EVPN support to extend pod network isolation outside the cluster.
+- No overlay mode to avoid the Geneve overhead.
+
+## User-Stories/Use-Cases
+
+- As a user, I want to be able to leverage my existing BGP network to dynamically
+  learn routes to pods in my Kubernetes cluster.
+- As a user, rather than having to maintain routes manually in each Kubernetes
+  node, as well as being constrained to using local gateway mode for respecting
+  user-defined routes; I want to use BGP so that I can dynamically advertise
+  egress routes for the Kubernetes pod traffic in either gateway mode.
+- As an egress IP user, I do not want to have to restrict my nodes to the same
+  layer 2 segment and prefer to use a pure routing implementation to handle
```

Comment:
Do we support advertising arbitrary EIP? I thought we decided to not support.

Reply:
We do? What do you mean with `arbitrary`? Is limited for L2 networks and VRF-Lite only.

Reply:
I meant EIPs that are not in the node IP subnet.

Reply:
Will clarify